openpype/hosts/maya/api/lib.py

```diff
@@ -2300,6 +2302,12 @@ def reset_frame_range(playback=True, render=True, fps=True):
         cmds.setAttr("defaultRenderGlobals.startFrame", animation_start)
         cmds.setAttr("defaultRenderGlobals.endFrame", animation_end)
 
+    project_name = get_current_project_name()
+    settings = get_project_settings(project_name)
+    if instances or settings["maya"]["update_publishable_frame_range"][
+            "enabled"]:
+        update_instances_frame_range()
```

Comment:
This won't work either - unfortunately, now there's no way to disable the behavior if user settings had it enabled. This is worse than what it was before unfortunately, since now resetting render settings will also reset all instances in the scene if it was enabled in settings - that's more confusing.

I think we have two options:
1. Revert to previous behavior, instances defaults to `True` but even when enabled only does something if it's enabled in settings and `False` always disables it.
2. Change `instances` default value to `None`, when `None` it uses the value from settings, when `True` it's force enabled, when `False` it's force disabled.

Better ideas are welcome of course.

Something like:

```python
if instances is None:
    # Allow default behavior to come from studio settings if True or False
    # is not explicitly set
    project_name = get_current_project_name()
    settings = get_project_settings(project_name)
    instances = settings["maya"]["update_publishable_frame_range"]["enabled"]

if instances:
    update_instances_frame_range()
```

Nice thing about the old behavior was sort of simplicity, if someone still wanted to reset just the instances no matter the setting you could still use: `update_instances_frame_range()` instead of `reset_frame_range`


@tokejepsen @antirotor Thoughts? Any preferences?

Reply:
I would opt for simpler solution of going with `instances=True` by default. Any forced update of the instances would happen through `update_instances_frame_range`

Reply:
I guess this would be much easier to resolve in new publisher where creators could handle that instead of "guessing" what and how should be updated?

Reply:
Not sure how the new publisher would solve this particular case. The issue here is mostly about something that is studio configurable behavior for `reset_frame_range` functionality (a menu entry in Maya too) and how it influences instances + how that functionality is accessible in the codebase.